chatkit/widgets.py

```diff
@@ -1002,36 +1087,22 @@ class LineSeries(BaseModel):
 Series = Annotated[
     BarSeries | AreaSeries | LineSeries,
     Field(discriminator="type"),
+    direct_usage_of_named_widget_types_deprecated,
 ]
 """Union of all supported chart series types."""
 
 
-class BasicRoot(WidgetComponentBase):
-    """Layout root capable of nesting components or other roots."""
-
-    type: Literal["Basic"] = Field(default="Basic", frozen=True)  # pyright: ignore
-    children: list[WidgetComponent | WidgetRoot]
-    """Children to render inside this root. Can include widget components or nested roots."""
-    theme: Literal["light", "dark"] | None = None
-    """Force light or dark theme for this subtree."""
-    direction: Literal["row", "col"] | None = None
-    """Flex direction for laying out direct children."""
-    gap: int | str | None = None
-    """Gap between direct children; spacing unit or CSS string."""
-    padding: float | str | Spacing | None = None
-    """Inner padding; spacing unit, CSS string, or padding object."""
-    align: Alignment | None = None
-    """Cross-axis alignment of children."""
-    justify: Justification | None = None
-    """Main-axis distribution of children."""
+class DynamicWidgetComponent(WidgetComponentBase):
+    """
+    A widget component with a statically defined base shape but dynamically
+    defined additional fields loaded from a widget template or JSON schema.
+    """
 
+    model_config = ConfigDict(extra="allow")
+    children: list["DynamicWidgetComponent"] | None = None
```

Comment:
`Transition` takes `DynamicWidgetComponent` not `list[DynamicWidgetComponent]`

```suggestion
    children: "DynamicWidgetComponent" | "list[DynamicWidgetComponent]" | None = None
```